codeforamerica/vita-min
# TBE-28 Allow transitioning intakes to all possible states when not in production

## [TBE-28](https://codeforamerica.atlassian.net/browse/TBE-28)
## Is PM acceptance required?
- Yes - don't merge until JIRA issue is accepted!
## What was done?
- Refactored all the transition code into a single location
- When not in production, state file admin users are permitted to transition an intake into any permitted next state.
- No change in production behaviour
- Offered alternative to PM where any notification could be triggered for any intake, but PM preferred this approach.
## How to test?
- Test on heroku.
- View a StateFileSubmission - you should see a new `Transition to State` section. (There is some overlap here between this and the existing buttons)
- Clicking on a state transitions the intake to that state.
- Risk Assessment
  - Change is isolated to non production environments, so risk is low.
## Screenshots (for visual changes)

### Non Production:
![image](https://github.com/codeforamerica/vita-min/assets/17094895/764ed4e3-4b3c-49d7-a055-22821255251c)

### Production
<img width="740" alt="image" src="https://github.com/codeforamerica/vita-min/assets/17094895/ac545a4b-bf58-4a8a-8b2f-896dda517b4f">


[TBE-28]: https://codeforamerica.atlassian.net/browse/TBE-28?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
